src/TorchSharp/Hub.cs

```diff
@@ -21,17 +22,46 @@ public static partial class hub
             /// <exception cref="InvalidDataException">SHA256 hash doesn't match</exception>
             public static void download_url_to_file(string url, string dst, string hash_prefix = null, bool progress = true)
             {
-                using (var webClient = new WebClient()) {
-                    if (progress) {
-                        // TODO: Implement progress.
-                        // Console.WriteLine("download_url_to_file(progress: true) is not supported.");
-                    }
-                    webClient.DownloadFile(url, dst);
-                    string fileHash = GetFileChecksum(dst);
-                    if (hash_prefix != null && !fileHash.StartsWith(hash_prefix)) {
-                        throw new InvalidDataException($"invalid hash value (expected \"{hash_prefix}\", got \"{fileHash}\"");
+                try {
+                    Task.Run(async () => {
+                        await download_url_to_file_async(url, dst, hash_prefix, progress);
+                    }).Wait();
+                } catch (AggregateException ex) {
+                    throw ex.InnerException;
+                }
+            }
+
+            /// <summary>
+            /// Download the url to a file 
+            /// </summary>
+            /// <param name="url">The URL to download</param>
+            /// <param name="dst">The file path to download the URL into</param>
+            /// <param name="hash_prefix">If non null, the SHA256 hash of downloaded content must match this prefix</param>
+            /// <param name="progress">Display the progress bar</param>
+            /// <exception cref="InvalidDataException">SHA256 hash doesn't match</exception>
+            public static async Task download_url_to_file_async(string url, string dst, string hash_prefix = null, bool progress = true)
```

Comment:
Should there be a cancellation token argument?

Reply:
Added `cancellationToken` to the arguments of `torch.hub.download_url_to_file_async()`.